docs/modules/links.md

```diff
@@ -0,0 +1,125 @@
+# Dynamic Links
+
+[Firebase Dynamic Links](https://firebase.google.com/docs/dynamic-links/) allows you to create and receive links on both Android and iOS platforms. Assuming the installation instructions have been followed, Firebase Dynamic Links is ready to go.
+
+
+All Dynamic Links operations are accessed via `firebase.links()`
+
+## Create Dynamic Links
+
+RNFirebase mimics [The REST API](https://firebase.google.com/docs/dynamic-links/rest) for Dynamic Links creation.
+The differences from the REST API are:
+1. The input for the methods is a javascript object instead of a JSON object.
+2. The response contains the URL string only.
+
+### Methods
+
+#### `createDynamicLink(parameters: Object): Promise<String>`
+
+Creates a long dynamic link.
+
+```javascript
+firebase.links().createDynamicLink({
+  dynamicLinkInfo: {
+    dynamicLinkDomain: "abc123.app.goo.gl",
+    link: "https://example.com/",
+    androidInfo: {
+      androidPackageName: "com.example.android"
+    },
+    iosInfo: {
+      iosBundleId: "com.example.ios"
+    }
+  }
+}).
+then((url) => {
+  // ...
+});
+```
+
+#### `createShortDynamicLink(parameters: Object): Promise<String>`
+
+Creates a short dynamic link.
+
+```javascript
+firebase.links().createShortDynamicLink(dynamicLinkInfo: {
+    dynamicLinkDomain: "abc123.app.goo.gl",
+    link: "https://example.com/",
+    androidInfo: {
+      androidPackageName: "com.example.android"
+    },
+    iosInfo: {
+      iosBundleId: "com.example.ios"
+    }
+  }
+}).
+then((url) => {
+  // ...
+});
+```
+### Parameters
+
+Only the following parameters are currently supported:
+
+```javascript
+{
+  dynamicLinkInfo: {
+    dynamicLinkDomain: string,
+    link: string,
+    androidInfo: {
+      androidPackageName: string,
+      androidFallbackLink: string,
+      androidMinPackageVersionCode: string,
+      androidLink: string
+    },
+    iosInfo: {
+      iosBundleId: string,
+      iosFallbackLink: string,
+      iosCustomScheme: string,
+      iosIpadFallbackLink: string,
+      iosIpadBundleId: string,
+      iosAppStoreId: string
+    },
+    socialMetaTagInfo: {
+      socialTitle: string,
+      socialDescription: string,
+      socialImageLink: string
+    }
+  },
+  suffix: {
+    option: "SHORT" or "UNGUESSABLE"
+  }
+}
+```
+?> Please note that when using `androidInfo` or `iosInfo`, `androidPackageName` and `iosBundleId` are mandatory (respectively).
+
+For more information [see reference](https://firebase.google.com/docs/reference/dynamic-links/link-shortener)
+
+## Receive Dynamic Links
+
+### Methods
+
+#### `getInitialLink(): Promise<String>`
+
+Call getInitialLink to access the URL that the app has been launched from. If the app was not launched from a URL, the return value is null.
+
+```javascript
+firebase.links().getInitialLink().then((url) => {
+  //...
+});
+```
+
+#### `onLink(listener: Function): () => any`
```

Comment:
Should we change this to:
```
onLink(listener: (String => void): () => void
```

Reply:
I don't know, it's also the same for messages, since the function is returned from react's event emitter.

Reply:
`onMessage()` doesn't specify a return value in the docs...